flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/ComputationState.java

```diff
@@ -114,6 +115,28 @@ public DeweyNumber getVersion() {
 		return version;
 	}
 
+	@Override
+	public boolean equals(Object obj) {
+		if (obj instanceof ComputationState) {
+			ComputationState other = (ComputationState) obj;
+			return Objects.equals(state, other.state) &&
+				event == other.event &&
```

Comment:
For `ComputationState`, I think two `ComputationState` equals only if the last taken events of them point to the same event. So `.equals()` may be not correct. Thoughts?